espnet2/asr/encoder/avhubert_encoder.py

```diff
@@ -0,0 +1,1092 @@
+# Copyright 2023
```

Comment:
Many thanks for making it working by adapting the original codebase of fairseq to here. My major concern is that the fairseq-style is not very compatible to our style (introducing a lot of module-related codes, which makes the design a bit redundant)

I feel that there would be two better solutions:
- Option1: follow https://github.com/ms-dot-k/espnet/blob/master/espnet2/asr/encoder/wav2vec2_encoder.py style that directly import fairseq?
- Option2: add an entry to s3prl (which we usually use for SSL and they have a good capacity of holding different SSL models) https://github.com/s3prl/s3prl

Reply:
I'm open for discussion. Maybe also consult @simpleoier and @sw005320 's thoughts on it

Reply:
I think it's better to copy it instead of importing it from fairseq.
We had a lot of issues when importing fairseq.
Also, this approach allows us to edit the model directly (e.g., adding adapter, etc.)